packages/aws-cdk/lib/archive.ts

```diff
@@ -24,12 +24,16 @@ export function zipDirectory(directory: string, outputFile: string): Promise<voi
     archive.on('error', fail);
     archive.pipe(output);
 
-    files.forEach(file => { // Append files serially to ensure file order
-      archive.append(fs.createReadStream(path.join(directory, file)), {
+    // Append files serially to ensure file order
+    for (const file of files) {
+      const fullPath = path.join(directory, file);
+      const [data, stat] = await Promise.all([fs.readFile(fullPath), fs.stat(fullPath)]);
```

Comment:
Again... I am concerned about memory usage. Can we throttle this some how? Instead of reading all files at once, can we read them one by one or in batches?

Reply:
The files are now read and appended one by one, this line just reads and "stats" in parallel a *single file*.

Reply:
Thanks for fixing this. It is currently crashing our build when zip files are being generated.